src/Http/Livewire/LivewireDatatable.php

```diff
@@ -1620,14 +1635,38 @@ public function checkboxQuery()
 
     public function toggleSelectAll()
     {
-        if (count($this->selected) === $this->getQuery()->getCountForPagination()) {
-            $this->selected = [];
+        $visible_chekboxes = $this->getQuery()->get()->pluck('checkbox_attribute')->toArray();
```

Comment:
typo: please call this variable `$visible_checkboxes`.

Reply:
Sorry, I just corrected it. You've a good eagle eye :smile: